spring-integration-core/src/main/java/org/springframework/integration/util/Pool.java

```diff
@@ -72,4 +72,11 @@
 	 */
 	int getAllocatedCount();
 
+	/**
+	 * Close the pool; returned items will be destroyed.
+	 * @since 4.3.23
+	 */
+	default void close() {
```

Comment:
The `default` is not going to work for Java 6, but we may not just introduce this method into the `Pool` interface for backward bytecode compatibility.

Reply:
Probably easiest to cast to `SimplePool` on 4.3.x?